frontend/src/app/modules/bookings/bookings-page/bookings-page.component.ts

```diff
@@ -143,7 +161,57 @@ export class BookingsPageComponent extends BaseComponent implements OnInit, OnDe
         return getDefaultTimeZone().timeValue;
     }
 
+    public currentDateChange() {
+        const meetingRequest = this.getMeetingRequest();
+
+        this.loadMeetings(meetingRequest);
+    }
+
+    private getMeetingRequest() {
+        let meetingRequest: IMeetingMembersRequest = { teamId: this.teamId };
+
+        if (this.selectedDateFilter === DateFilterValue.Today) {
+            meetingRequest = {
+                ...meetingRequest,
+                start: moveByTimezone(new Date(this.currentDate.getFullYear(), this.currentDate.getMonth(), this.currentDate.getDate())),
+                end: moveByTimezone(new Date(this.currentDate.getFullYear(), this.currentDate.getMonth(), this.currentDate.getDate() + 1)),
+            };
+        }
+
+        if (this.selectedDateFilter === DateFilterValue.Past) {
+            meetingRequest = {
+                ...meetingRequest,
+                end: moveByTimezone(new Date()),
+            };
+        }
+
+        if (this.selectedDateFilter === DateFilterValue.Pending) {
+            meetingRequest = {
+                ...meetingRequest,
+                start: moveByTimezone(new Date()),
+            };
+        }
+
+        if (this.selectedDateFilter === DateFilterValue.Range) {
+            if (!this.currentEnd || this.currentStart.getTime() === this.currentEnd.getTime()) {
+                this.currentEnd = addDays(this.currentStart, 1);
+            }
+            meetingRequest = {
+                ...meetingRequest,
+                start: moveByTimezone(this.currentStart),
+                end: moveByTimezone(this.currentEnd),
+            };
+        }
+
+        return meetingRequest;
+    }
+
     private getNumberOfItemsToDisplay(width: number) {
+        if (width > widthToContainFiveItems) {
+            this.numberOfMembersToDisplay = 5;
+
+            return;
+        }
```

Comment:
It would be better to know the size of one item and dynamic calculate items' count